### ISSUE TITLE
Difficulty getting flac encoding working.

### ISSUE THREAD

**Issue by lschapker** [JoeLametta/whipper#118]
I am extremely new to "whipper" and python in general (been using Ubuntu for quite some time and am familiar with debugging Java programs).

After a bit of work, I have Whipper able to find and use my disk drive and all the initial setup appears to be configured (i.e. all the dependencies I know).

I go to rip my first CD, and I see whipper making 5 attempts to encode the first track (using Flac) failing with "ERROR: cannot find flac plugin" and then failing overall with "... track can't be ripped. ...".  I have verified that flac is installed (/usr/bin/flac).

I've had to do quite a bit of research and trial and error to get this far (i.e. figuring out what packages to install into my distribution {Ubuntu 16.10}) and I believe that I'm close to cracking this nut.  I'm not finding anything to sink my teeth into to be able to debug this further (i.e. no compile errors from the python scripts).

Can someone please provide some guidance as to how I might go about debugging this further?

Thank you in advance.

**Comment by lschapker** [comment]
Here is the log file from an attempted rip.

What does Error: no element "decodebin" mean?

[whipper.zip](https://github.com/JoeLametta/whipper/files/736709/whipper.zip)

**Comment by JoeLametta** [comment]
> Can someone please provide some guidance as to how I might go about debugging this further?

Here are the dependencies needed for Ubuntu 16.04 (courtesy of @inverse):

```bash
# Prerequisites
sudo apt install build-essential python-dev python-pip libcdio-dev libiso9660-dev swig pkg-config

# Core components
sudo apt install  cdparanoia cdrdao libgstreamer1.0 python-gst0.10 python-musicbrainzngs python-setuptools python-cddb libsndfile1-dev flac sox

# Pip components
pip install pycdio --user
```

Unfortunately, right now, it seems you can't get a complete installation on Ubuntu 16.10: please see [this one](https://github.com/JoeLametta/whipper/pull/110#issuecomment-273550126).
That happens because whipper still relies on GStreamer 0.10 which has long been deprecated and, recently, it's being removed from modern distributions. We've planned to drop that dependency long ago and that's hopefully going to be completed soon [issue here](https://github.com/JoeLametta/whipper/issues/29#issuecomment-275455166).

> Thank you in advance.

You're welcome!

**Comment by lschapker** [comment]
Just as an FYI to anyone in the future...

On Unbuntu 16.10, I had most of the packages that JoeLametta had stated (but was missing a couple).  After the addition, I was still receiving the "ERROR: cannot find flac plugin" message.  After a long night 'stepping through the code', I found the reason:  the "gstreamer0.10-plugins-good" package was not installed.

I downloaded the following packages from "http://packages.ubuntu.com/xenial/amd64":
gstreamer0.10-plugins-good_0.10.31-3+nmu4ubuntu2.16.04.2_amd64.deb
gstreamer0.10-plugins-base_0.10.36-2_amd64.deb
libpng12-0_1.2.54-1ubuntu1_amd64.deb
(I had previously downloaded)
libgstreamer0.10-0_0.10.36-1.5_amd64.deb
libgstreamer-plugins-base0.10-0_0.10.36-2_amd64.deb
and installed them using "sudo dpkg -i <list of DEB files>".

I now have Whipper running on Ubuntu 16.10.

Thank you for your assistance!
